instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/trace/SpanE2ETests.kt

```diff
@@ -0,0 +1,196 @@
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#setTag(String, String)
+     */
+    @Test
+    fun trace_span_set_tag_string() {
+        val testMethodName = "trace_span_set_tag_string"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .setTag(SPECIAL_STRING_TAG_NAME, "str${forge.anAlphaNumericalString()}")
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#setTag(String, Number)
+     */
+    @Test
+    fun trace_span_set_tag_number() {
+        val testMethodName = "trace_span_set_tag_number"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .setTag(
+                SPECIAL_NUMBER_TAG_NAME,
+                forge.anElementFrom(
+                    forge.aLong(1, 10),
+                    forge.anInt(1, 10),
+                    forge.aFloat(1f, 10f),
+                    forge.aDouble(1.0, 10.0),
+                    forge.anInt(1, 10).toByte()
+                )
+            )
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#<T>setTag(Tag<T>, T)
+     */
+    @Test
+    fun trace_span_set_tag_generic() {
+        val testMethodName = "trace_span_set_tag_generic"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .setTag(StringTag(SPECIAL_GENERIC_TAG_NAME), "str${forge.anAlphaNumericalString()}")
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#setBaggageItem(String, String)
+     */
+    @Test
+    fun trace_span_set_baggage_item() {
+        val testMethodName = "trace_span_set_baggage_item"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .setBaggageItem(SPECIAL_STRING_TAG_NAME, "str${forge.anAlphaNumericalString()}")
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#log(Map)
+     */
+    @Test
+    fun trace_span_log_map() {
+        val testMethodName = "trace_span_log_map"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .log(
+                mapOf(
+                    "${SPECIAL_NUMBER_TAG_NAME}_positive" to forge.aSmallInt(),
+                    "${SPECIAL_NUMBER_TAG_NAME}_negative" to -forge.aSmallInt(),
+                    TEST_METHOD_NAME_KEY to testMethodName
+                )
+            )
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#log(String)
+     */
+    @Test
+    fun trace_span_log_string() {
+        val testMethodName = "trace_span_log_string"
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .log("${testMethodName}_str${forge.anAlphaNumericalString()}")
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Span#log(long, Map)
+     */
+    @Test
+    fun trace_span_log_timestamp_map() {
```

Comment:
I guess you could add more test cases here for different types following what I did in the `LoggerE2ETests` but it is not mandatory as everything is tested there already.

Reply:
I was thinking `string` + `int` will be enough here, since it is the types supported on DD. Maybe makes sense to add `double` as well though.

Reply:
added `double` values for the assertion as well.